.NET/Microsoft.Recognizers.Text/Matcher/AcAutomaton.cs

```diff
@@ -5,16 +5,15 @@ namespace Microsoft.Recognizers.Text.Matcher
 {
     public class AcAutomaton<T> : AbstractMatcher<T>
     {
-        protected readonly AaNode<T> root = new AaNode<T>();
```

Comment:
Let's extract this change to the same PR ad the one hiding the field. (They are similar changes, breaking changes because we are renaming a semi-public property: anyone with working code using this field in a child class will get an error)